codebase/git_test.go

```diff
@@ -0,0 +1,36 @@
+package codebase_test
+
+import (
+	"testing"
+
+	"github.com/fabric8-services/fabric8-common/codebase"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGitURL(t *testing.T) {
+	tables := []struct {
+		url     string
+		target  codebase.Scheme
+		want    string
+		wantErr bool
+	}{
+		{"git@github.com:testuser/testrepo.git", codebase.HTTPS, "https://github.com/testuser/testrepo.git", false},
+		{"https://github.com/testuser/testrepo.git", codebase.HTTPS, "https://github.com/testuser/testrepo.git", false},
+		{"git@github.com:testuser/testrepo", codebase.HTTPS, "", true},
+		{"git@anything/testrepo", codebase.HTTPS, "", true},
+		{"anything/testrepo", codebase.HTTPS, "", true},
+		{"http://github.com/testuser/testrepo.git", codebase.HTTPS, "https://github.com/testuser/testrepo.git", false},
+		{"http://anything/testrepo.git", codebase.HTTPS, "", true},
+	}
```

Comment:
Added test to check from_scheme support `http`, `git/ssh`.  Also added test with invalid URLs.